pkg/helm/helm.go

```diff
@@ -87,14 +86,11 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 }
 
 func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
-	var values map[string]interface{}
-	err := json.Unmarshal([]byte(h.Values), &values)
-	if err != nil {
-		return []string{}, fmt.Errorf("failed to unmarshal helm.values, error: %w", err)
-	}
-	for key, val := range values {
-		args = append(args, "--set", fmt.Sprintf("%s=%v", key, val))
+	valuesPath := filepath.Join(os.TempDir(), "chart-values.yaml")
+	if err := os.WriteFile(valuesPath, []byte(h.Values), 0644); err != nil {
+		return []string{}, fmt.Errorf("failed to create values file: %w", err)
```

Comment:
nit: return `nil` instead of initializing`[]string{}`.

Reply:
Sounds good.